common/filestore_utils.py

```diff
@@ -27,16 +31,35 @@ def _using_gsutil():
     return experiment_filestore_path.startswith('gs://')
 
 
-if _using_gsutil():
-    from common import gsutil as filestore_utils_impl
-else:
-    # When gsutil is not used in the context, here it should use local_utils.
-    # TODO(zhichengcai): Implement local_filestore.py and import it here.
-    from common import gsutil as filestore_utils_impl
+global filestore_utils_impl
+filestore_utils_impl = None
+
+
+def get_filestore_utils_impl():
+    """Imports filestore_utils_impl dynamically."""
+
```

Comment:
Btw I don't think you made this decorator quite right. You should probably only have one layer of nested functions not two. That's why your decorators need to call get_filestore_utils_impl() which is very weird for a decorator.

Reply:
I tried two layers, but it doesn't work. This is the only workable version. :)
If only two layers, it reports for arguments amount conflict.